tapchannel/aux_sweeper.go

```diff
@@ -0,0 +1,1475 @@
+		signingKey, leafToSign := applySignDescToVIn(
+			signDesc, vIn, &a.cfg.ChainParams,
+			scriptTree.TapTweak(),
+		)
+
+		// In this case, the witness isn't special, so we'll set the
+		// control block now for it.
+		vIn.TaprootLeafScript[0].ControlBlock = ctrlBlkBytes
+
+		log.Infof("signing vPacket for input=%v",
+			spew.Sdump(vIn.PrevID))
+
+		// With everything set, we can now sign the new leaf we'll
+		// sweep into.
+		signed, err := a.cfg.Signer.SignVirtualPacket(
+			vPacket, tapfreighter.SkipInputProofVerify(),
+			tapfreighter.WithValidator(&schnorrSigValidator{
+				pubKey:     signingKey,
+				tapLeaf:    lfn.Some(leafToSign),
+				signMethod: input.TaprootScriptSpendSignMethod,
+			}),
+		)
+		if err != nil {
+			return fmt.Errorf("error signing virtual "+
+				"packet: %w", err)
+		}
+
+		if len(signed) != 1 || signed[0] != 0 {
+			return fmt.Errorf("error signing virtual packet, " +
+				"got no sig")
+		}
+	}
+
+	return nil
+}
+
+// createAndSignSweepVpackets creates vPackets that sweep the funds from the
+// channel to the wallet, and then signs them as well.
+func (a *AuxSweeper) createAndSignSweepVpackets(
+	sweepInputs []*cmsg.AssetOutput, signDesc input.SignDescriptor,
+	sweepDesc lfn.Result[tapscriptSweepDesc],
+) lfn.Result[[]*tappsbt.VPacket] {
+
+	// Based on the sweep inputs, make vPackets that sweep all the inputs
+	// into a new output with a fresh script key. They won't have an
+	// intenral key set, we'll do that when we go to make the output to
+	// anchor them all. We'll then take those, then sign all the vPackets
+	// based on the specified sweepDesc.
+	signPkts := func(vPkts []*tappsbt.VPacket,
+		desc tapscriptSweepDesc) lfn.Result[[]*tappsbt.VPacket] {
+
+		err := a.signSweepVpackets(
+			vPkts, signDesc, desc.scriptTree, desc.ctrlBlockBytes,
+		)
+		if err != nil {
+			return lfn.Err[[]*tappsbt.VPacket](err)
+		}
+
+		return lfn.Ok(vPkts)
+	}
+
+	return lfn.AndThen2(
+		a.createSweepVpackets(sweepInputs), sweepDesc, signPkts,
+	)
+}
+
+// tapscriptSweepDesc is a helper struct that contains the tapscript tree and
+// the control block needed to generate a valid spend.
+//
+// TODO(roasbeef): only needs the merkle root?
+type tapscriptSweepDesc struct {
+	scriptTree input.TapscriptDescriptor
+
+	ctrlBlockBytes []byte
+}
+
+// commitNoDelaySweepDesc creates a sweep desc for a commitment output that
+// resides on the remote party's commitment transaction. This output is a
+// non-delay output, so we don't need to worry about the CSV delay when
+// sweeping it.
+func commitNoDelaySweepDesc(keyRing *lnwallet.CommitmentKeyRing,
+) lfn.Result[tapscriptSweepDesc] {
+
+	// We'll make the script tree for the to remote script (we're remote as
+	// this is their commitment transaction). We don't have an auxLeaf here
+	// as we're on the TAP layer.
+	toRemoteScriptTree, err := input.NewRemoteCommitScriptTree(
+		keyRing.ToRemoteKey, input.NoneTapLeaf(),
+	)
+	if err != nil {
+		return lfn.Err[tapscriptSweepDesc](err)
```

Comment:
I think we should enhance the error message here so that we can be sure to pin point its origin from the logs. Other similar opportunities in this file.